### PR TITLE
Unify clientset generation

### PR DESCRIPTION
**Pull request by twz123**
## Description

* Add doc.go to APIs
  So that the k8s code-generator can parse the tags correctly.

* Remove k0sproject.io from API package folder names and generate one clientset for all APIs
  This makes paths and imports more concise and resembles the canonical way of naming API folders and structuring clientsets in other projects that use k8s code generation.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [x] Manual test
- [ ] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pkg/apis/autopilot/doc.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package autopilot contains API Schema definitions for the
+// autopilot.k0sproject.io API group.
+package autopilot
+
+const GroupName = "autopilot.k0sproject.io"
```
Comment (mikhail-sakhnov) [inline pkg/apis/autopilot/doc.go:19]: I wonder why linter is not failing with a public constant without a comment. Is it not required for doc.go?

Comment (twz123) [reply pkg/apis/autopilot/doc.go:19]: I don't think that there's a lint for that. There are quite a bunch of public constants without docs, e.g. in the constant package.

Comment (kke) [reply pkg/apis/autopilot/doc.go:19]: [Revive](https://github.com/mgechev/revive) is enabled, it should nag about these. 🤔

Comment (twz123) [reply pkg/apis/autopilot/doc.go:19]: What's the [rule name](https://revive.run/docs#available-rules)? I only see a rule about package comments.

Comment (kke) [reply pkg/apis/autopilot/doc.go:19]: Ah, EXC0011 is in default exclusions list and should be enabled separately for that to happen.

**Comment by twz123** [comment]
See #2909.

File: pkg/apis/autopilot/doc.go (hunk 1 above)
Comment (juanluisvaladas) [reply pkg/apis/autopilot/doc.go:19]: Actually I think there is indeed a problem with revive. If I run golangci-lint in my machine I get some extra errors.

I filed an issue upstream in golangci-lint, I also need to investigate this more locally to see if there is a difference.
https://github.com/golangci/golangci-lint/issues/3764

**Review by juanluisvaladas — DISMISSED** [review]

**Comment by twz123** [comment]
This constantly hits #3038.

**Review by juanluisvaladas — APPROVED** [review]
Approving again, this time without further review because there were no merging conflicts

**Pull request merged**